app/models/shift.rb

```diff
@@ -2,4 +2,26 @@ class Shift < ActiveRecord::Base
   belongs_to :giver, class_name: :User, foreign_key: :giver_id
   belongs_to :taker, class_name: :User, foreign_key: :taker_id
   has_many :offers
+
+  has_one :team, through: :giver, source: :team
+
+  def yes_offers?
```

Comment:
While this method returns a boolean value, it also has a side-effect (setting the value of `self.covered`), which is the real purpose of the method. I suggest ending the name with a `!` to indicate that fact, rather than `?`. Perhaps something like `determine_offer_status!`.